prody/proteins/interactions.py

```diff
@@ -0,0 +1,1109 @@
+    :type atoms: :class:`.Atomic`
+    
+    :arg distA: non-zero value, maximal distance between ion and residue.
+    :type distA: int or float, default is 3.0.
+    
+    :arg extraIons: ions to be included in the analysis.
+    :type extraIons: list of str
+    
+    :arg excluded_ions: ions which should be excluded from the analysis.
+    :type excluded_ions: list of str """
+    
+    try:
+        coords = (atoms._getCoords() if hasattr(atoms, '_getCoords') else
+                    atoms.getCoords())
+    except AttributeError:
+        try:
+            checkCoords(coords)
+        except TypeError:
+            raise TypeError('coords must be an object '
+                            'with `getCoords` method')
+    
+    try:
+        atoms_ions = atoms.select('ion and not name '+' '.join(excluded_ions)+' or (name '+' '.join(map(str,extraIons))+')')
+        MetalResList = []
+        MetalRes_calculations = cleanNumbers(findNeighbors(atoms_ions, distA, atoms.select('all and noh')))
+        for i in MetalRes_calculations:
+            if i[-1] != 0:
+                MetalResList.append([atoms.getResnames()[i[0]]+str(atoms.getResnums()[i[0]]), i[2], 
+                                 atoms.getResnames()[i[1]]+str(atoms.getResnums()[i[1]]), i[3], i[-1]])
+
+        return MetalResList
+        
+    except TypeError:
+        raise TypeError('An object should contain ions')
+
+
+def calcProteinInteractions(atoms):
+    """Compute all protein interactions (shown below) using default paramaters.
+        (1) Hydrogen bonds
+        (2) Salt Bridges
+        (3) RepulsiveIonicBonding 
+        (4) Pi stacking interactions
+        (5) Pi-cation interactions
+        (6) Hydrophobic interactions
+    
+    :arg atoms: an Atomic object from which residues are selected
+    :type atoms: :class:`.Atomic`"""
+
+    try:
+        coords = (atoms._getCoords() if hasattr(atoms, '_getCoords') else
+                    atoms.getCoords())
+    except AttributeError:
+        try:
+            checkCoords(coords)
+        except TypeError:
+            raise TypeError('coords must be an object '
+                            'with `getCoords` method')
+
+    LOGGER.info('Calculating all interations.') 
+    HBs_calculations = calcHydrogenBonds(atoms.protein)               #1 in scoring
+    SBs_calculations = calcSaltBridges(atoms.protein)                 #2
+    SameChargeResidues = calcRepulsiveIonicBonding(atoms.protein)     #3
+    Pi_stacking = calcPiStacking(atoms.protein)                       #4
+    Pi_cation = calcPiCation(atoms.protein)                           #5
+    Hydroph_calculations = calcHydrophohic(atoms.protein)             #6
+    AllInteractions = [HBs_calculations, SBs_calculations, SameChargeResidues, Pi_stacking, Pi_cation, Hydroph_calculations]   
+    
+    return AllInteractions
+    
+
+def calcInteractionMatrix(atoms):
+    """Calculate matrix with protein interactions which is scored by as follows:
```

Comment:
remove by

Reply:
You should probably explain the origin of these scores/weights and give users the option to change them like with the gap and mismatch scores for sequence alignment.